c_src/membrane_ffmpeg_swscale_plugin/pixel_format_converter/converter.c

```diff
@@ -0,0 +1,120 @@
+#include "converter.h"
+#include <libavutil/error.h>
+#include <libavutil/frame.h>
+#include <libavutil/imgutils.h>
+#include <libswscale/swscale.h>
+
+enum AVPixelFormat string_to_AVPixelFormat(char *);
+
+UNIFEX_TERM create(UnifexEnv *env, uint64_t width, uint64_t height,
+                   char *old_format, char *new_format) {
+  UNIFEX_TERM res;
+  const enum AVPixelFormat input_fmt = string_to_AVPixelFormat(old_format),
+                           output_fmt = string_to_AVPixelFormat(new_format);
+
+  if (input_fmt == AV_PIX_FMT_NONE) {
+    return create_result_error(env, "unknown_input_format");
+  } else if (!sws_isSupportedInput(input_fmt)) {
+    return create_result_error(env, "unsupported_input_format");
+  }
+
+  if (output_fmt == AV_PIX_FMT_NONE) {
+    return create_result_error(env, "unknown_output_format");
+  } else if (!sws_isSupportedOutput(output_fmt)) {
+    return create_result_error(env, "unsupported_output_format");
+  }
+
+  State *state = unifex_alloc_state(env);
+  state->sws_context =
+      sws_getContext(width, height, input_fmt, width, height, output_fmt,
+                     SWS_BICUBIC, NULL, NULL, NULL);
+
+  if (state->sws_context == NULL) {
+    res = create_result_error(env, "create_context_failed");
+    goto end;
+  }
+
+  state->width = width;
+  state->height = height;
+  state->srcFormat = input_fmt;
+  state->dstFormat = output_fmt;
+
+  // Allocate memory for both the input and output frames
+  int src_image_size =
+      av_image_alloc(state->src_data, state->src_linesize, state->width,
+                     state->height, state->srcFormat, 1);
+  state->dst_image_size =
+      av_image_alloc(state->dst_data, state->dst_linesize, state->width,
+                     state->height, state->dstFormat, 1);
```

Comment:
It is really easy to add the alignment to this converter - by replacing the `1` here with `32` The only other difference is that instead of keeping `dst_image_size` as a size of the buffer allocated here, you need to call `av_image_get_buffer_size` with alignment `1`. The `av_image_copy_to_buffer` will handle a copy from aligned `av_image` to a non-aligned buffer